hardware/MQTT.cpp

```diff
@@ -278,6 +278,57 @@ void MQTT::on_message(const struct mosquitto_message *message)
 		}
 		return;
 	}
+	else if (szCommand == "setcolbrightnessvalue")
+	{
+		idx = (uint64_t)root["idx"].asInt64();
+		
+		if (root["switchcmd"].empty())
+			root["switchcmd"] = "On";
+		if (!root["switchcmd"].isString())
+			goto mqttinvaliddata;
+			
+		std::string switchcmd = root["switchcmd"].asString();
+		if ((switchcmd != "On") && (switchcmd != "Off") && (switchcmd != "Toggle") && (switchcmd != "Set Level"))
+			goto mqttinvaliddata;
+			
+		int level = 0;
+		if (!root["level"].empty())
+		{
+			if (root["level"].isString())
+				level = atoi(root["level"].asString().c_str());
+			else
+				level = root["level"].asInt();
+		}
+		
+		int hue = 0;
+		if (!root["hue"].empty())
+		{
+			if (root["hue"].isString())
+				hue = atoi(root["hue"].asString().c_str());
+			else
+				hue = root["hue"].asInt();
+		}
+		
+		bool isWhite = false;
+		if (!root["isWhite"].empty())
+		{
+			if (root["isWhite"].isString())
+				isWhite = atoi(root["isWhite"].asString().c_str());
+			else
+				isWhite = root["isWhite"].asInt();
+		}
+		
+		if (!m_mainworker.SwitchLight(idx, switchcmd, level, hue, isWhite, 0) == true)
+		{
+			_log.Log(LOG_ERROR, "MQTT: Error sending switch command!");
```

Comment:
actually those "extra" log lines where supposed to have been removed, they where only there for debug :-)
done now